### ISSUE TITLE
Latest version does not work at all

### ISSUE THREAD

**Issue by richardhj** [heimrichhannot/contao-ldap#1]
Hello, I installed your latest stable version but various errors and warnings encountered for me. No login is possible at all.
I aim to use ldap for the back end users.

```
Warning: Declaration of HeimrichHannot\Ldap\LdapPersonModel::findAll() should be compatible with Contao\Model::findAll(array $arrOptions = Array) in composer/vendor/heimrichhannot/contao-ldap/models/LdapPersonModel.php on line 5
#0 composer/vendor/heimrichhannot/contao-ldap/models/LdapPersonModel.php(5): __error(2, 'Declaration of ...', '/Users/richard/...', 5, Array)
#1 composer/vendor/terminal42/contao-namespace-class-loader/library/Composer/Autoload/ClassLoader.php(363): include('/Users/richard/...')
#2 composer/vendor/terminal42/contao-namespace-class-loader/library/Composer/Autoload/ClassLoader.php(269): Composer\Autoload\includeFile('/Users/richard/...')
#3 [internal function]: Composer\Autoload\ClassLoader->loadClass('HeimrichHannot\\...')
#4 composer/vendor/heimrichhannot/contao-ldap/models/LdapUserModel.php(5): spl_autoload_call('HeimrichHannot\\...')
#5 composer/vendor/terminal42/contao-namespace-class-loader/library/Composer/Autoload/ClassLoader.php(363): include('/Users/richard/...')
#6 composer/vendor/terminal42/contao-namespace-class-loader/library/Composer/Autoload/ClassLoader.php(269): Composer\Autoload\includeFile('/Users/richard/...')
#7 [internal function]: Composer\Autoload\ClassLoader->loadClass('HeimrichHannot\\...')
#8 composer/vendor/heimrichhannot/contao-ldap/classes/backend/LdapPerson.php(56): spl_autoload_call('HeimrichHannot\\...')
#9 composer/vendor/heimrichhannot/contao-ldap/classes/backend/LdapPerson.php(21): HeimrichHannot\Ldap\Backend\LdapPerson::authenticateLdapPerson('uesrname...', '***')
#10 system/modules/core/library/Contao/User.php(324): HeimrichHannot\Ldap\Backend\LdapPerson->importPersonFromLdap('uesrname...', '***', 'tl_user')
#11 system/modules/core/controllers/BackendIndex.php(37): Contao\User->login()
#12 contao/index.php(19): Contao\BackendIndex->__construct()
#13 {main}
```
There are various parameter incompatible function calls in the code base. I would like to contribute with PRs but it seems like your refactoring is not done yet.

```
Fatal error: Uncaught exception Exception with message Please fully configure your LDAP connection at first. thrown in composer/vendor/heimrichhannot/contao-ldap/classes/Ldap.php on line 50
#0 composer/vendor/heimrichhannot/contao-ldap/classes/Ldap.php(61): HeimrichHannot\Ldap\Ldap::connect(true)
#1 composer/vendor/heimrichhannot/contao-ldap/models/LdapPersonModel.php(51): HeimrichHannot\Ldap\Ldap::getConnection(true)
#2 composer/vendor/heimrichhannot/contao-ldap/classes/backend/LdapPerson.php(56): HeimrichHannot\Ldap\LdapPersonModel::findByUsername('uesrname...')
#3 composer/vendor/heimrichhannot/contao-ldap/classes/backend/LdapPerson.php(21): HeimrichHannot\Ldap\Backend\LdapPerson::authenticateLdapPerson('uesrname...', '***')
#4 system/modules/core/library/Contao/User.php(324): HeimrichHannot\Ldap\Backend\LdapPerson->importPersonFromLdap('uesrname...', '***', 'tl_user')
#5 system/modules/core/controllers/BackendIndex.php(37): Contao\User->login()
#6 contao/index.php(19): Contao\BackendIndex->__construct()
#7 {main}
```
This occurs because `Ldap::connect()` expects `$strMode` to be either `user` or `member` but the variable is boolean true for back end mode.

**Comment by Defcon0** [comment]
Hello,

there was large amounts of refactoring going on in v2. Seems like some code didn't make it in the repo. I'll take a look tomorrow.

**Comment by richardhj** [comment]
Could you please add `heimrichhannot/contao-multi_column_editor` to your dependencies in addition? :)

**Comment by Defcon0** [comment]
OK, I just fixed the mode issue and the composer dependency. Also I made clear in the README that the module currently is tied to a certain ldap implementation, i.e. the way data is stored there.

I described the problem in the first paragraph of the README. At the moment we don't have enough time to implement the flexible version. But you are of course very welcome to come up with a PR solving that problem.
